### ISSUE TITLE
Disable CanvasBlocker first-run page

### ISSUE THREAD

**Issue by allo-** [allo-/firefox-profilemaker#270]
CanvasBlocker now supports disabling the first-run page: https://github.com/kkapsner/CanvasBlocker/issues/656

Also `"extensions.CanvasBlocker@kkapsner.de.whiteList": ""` should be reviewed if it is needed / useful.

**Comment by allo-** [comment]
Waiting for 1.10 on [AMO](https://addons.mozilla.org/en-US/firefox/addon/canvasblocker/versions/) for a short test before adding it.

**Comment by kkapsner** [comment]
You could test with the alpha version: https://canvasblocker.kkapsner.de/versions/

**Comment by kkapsner** [comment]
The default value for whitelist is the empty string since six years: https://github.com/kkapsner/CanvasBlocker/blob/master/lib/settingDefinitions.js#L47

**Comment by allo-** [comment]
Is it signed with the right key for AMO or do I need to use a nightly to install the alpha version? The zip download bundles the xpi and I think the right signature would be enough, but I am not sure about the enterprise policy version. In the end, enterprise policies are the cleanest way to install addons by default, but they require the user to modify the installation directory instead of the profile directory and affect all profiles.

**Comment by kkapsner** [comment]
It is properly signed but self hosted. I have it installed in my daily Firefox profile (not nightly).

I have no clue about the enterprise policy version.

**Comment by allo-** [comment]
Looks like for this type of settings yet another settings store needs to be implemented. Most settings are settings that afaik aren't used by extensions anymore and simple inline files, e.g., for [settings as JS file](https://github.com/allo-/firefox-profilemaker/blob/38d964dd10fc22f78ba38ff834fff4034d0ff605/settings/privacybadger.json#L3) are implemented, but I think the settings here are stored in indexedDB?

I'm not sure what's the best way to write them from outside Firefox. Current Firefox seems to store them as sqlite3, but a dump doesn't contain the settings in a readable format.

**Comment by kkapsner** [comment]
Yes - the storage of the webExtension settings is different to old legacy settings and it also changed over time.

Maybe you can work with the file `browser-extension-data/CanvasBlocker@kkapsner.de/storage.js` and the preference (in pref.js)
```JS
user_pref("extensions.webextensions.ExtensionStorageIDB.migrated.CanvasBlocker@kkapsner.de", false);
```

Not tested at all and a shot in the dark but worth a try because it's quite simple.

**Comment by allo-** [comment]
I already see the migration being the next thing that is removed by Mozilla.

What I do not understand is that the sqlite3 file contains the settings names as text in the binary, but a database dump does not contain these strings.

**Comment by allo-** [comment]
I guess with the requirement to create the storage in the right way it does no longer count as easy. Or may become easy again with #271.

**Comment by allo-** [comment]
It would be particularly nice to implement it for temporary/disposable profiles, which open the first run tab each time. With a good implementation of #271 other addons with first run pages would also become more feasible.
